core/payment-driver/zksync/src/service.rs

```diff
@@ -88,6 +113,22 @@ async fn schedule_payment(
         amount: msg.amount(),
         date: Some(Utc::now()),
     };
+
+    let pub_address = Address::from_str(&details.sender[2..]).unwrap();
+    let seed = get_zksync_seed(pub_address).await;
+    let provider = RpcClient::new(ZKSYNC_RPC_ADDRESS);
+    let wallet = Wallet::from_seed(seed, pub_address, provider);
+
+    let recepient = Address::from_str(&details.recipient[2..]).unwrap();
+    let amount = BigUint::from_str(&details.amount.to_string()).unwrap();
+    let amount = amount * pow(BigUint::from(10u32), 18);
```

Comment:
Looks like a candidate for a constant.
Or, if we want to be super-clean, it should use the `decimals` parameter from zkSync token list (https://api.zksync.io/api/v0.1/tokens).

Reply:
Like this idea!
Added a TODO for the second option, since it also needs a update to the rust crate